src/main/scala/org/clulab/wm/eidos/exporters/DebugGroundingExporter.scala

```diff
@@ -1,41 +1,175 @@
+          val info = args.map(m => (m.odinMention.text, m.grounding.get("wm_compositional"), m.grounding.get("wm_flattened")))
+          for ((text, groundingComp, groundingFlat) <- info) {
+            if (text=="famine") {
+              //do nothing
+            }
+            else {
+              pw.println(s"mention text: ${text}\n")
+              pw.println(s"mention entities: ${args.head.odinMention.entities.get}\t")
+
+              pw.println(s"Flat Grounding:")
+              groundingFlat match {
+                case None => pw.println("no grounding...")
+                case Some(grounding) =>
+                  grounding.grounding.foreach { gr =>
+                    pw.println(s"  --> grounding:  ${gr.name}")
+                    pw.println(s"      score:      ${gr.score}")
+                  }
               }
-          }
-          pw.println(s"Flat Grounding:")
-          groundingFlat match {
-            case None => pw.println("no grounding...")
-            case Some(grounding) =>
-              grounding.grounding.foreach { gr =>
-                pw.println(s"  --> grounding: ${gr.name}")
-                pw.println(s"      score: ${gr.score}")
+              pw.println()
+
+              pw.println(s"Compositional Grounding:")
+              groundingComp match {
+                case None => pw.println("no grounding...")
+                case Some(grounding) =>
+                  var tupNum: Int = 0
+                  grounding.grounding.foreach { gr =>
+                    pw.println(s"Tuple $tupNum")
+                    tupNum += 1
+                    val tuple = gr.asInstanceOf[PredicateGrounding].predicateTuple
+                    val labels = Seq("Theme", "Theme Prop", "Process", "Process Prop")
+                    val slots = Seq(tuple.theme, tuple.themeProperties, tuple.themeProcess, tuple.themeProcessProperties)
+//                    val theme_name = if (tuple.theme.nonEmpty) slots.head.grounding.head.name else None
+                    for (j <- labels.indices) {
+                      pw.println(s"\t${labels(j)}")
+                      for (g <- slots(j).grounding) {
+//                        if (g.name == theme_name && j != 0) {
+//                          //do nothing
+//                        }
+//                        else {
+//                          pw.println(s"    --> grounding ${j}:   ${g.name}")
+//                          pw.println(s"        score:        ${g.score}")
+//                        }
+                        pw.println(s"    --> grounding ${j}:   ${g.name}")
+                        pw.println(s"        score:        ${g.score}")
+                        //                      pw.println()
+                      }
+                      pw.println()
+
+                    }
+                  }
               }
+            }
+
+            }
+
           }
-          pw.println()
-        }
 
       }
     }
+
+
+
+
```

Comment:
I think we should prune some of this commented out stuff unless you need it

Reply:
Sounds good to me.